telemetry/derive/src/lib.rs

```diff
@@ -0,0 +1,253 @@
+//! Attribute-like macro for instrumenting `isi` for `prometheus`
+//! metrics. See [`metrics`] for more details.
+
+use proc_macro::TokenStream;
+#[cfg(feature = "metric-instrumentation")]
+use proc_macro2::TokenStream as TokenStream2;
+use proc_macro_error::{abort, proc_macro_error};
+use quote::{quote, ToTokens};
+use syn::{
+    parse::Parse, parse_macro_input, punctuated::Punctuated, token::Comma, FnArg, ItemFn, LitStr,
+    Path, Type,
+};
+
+// TODO: export these as soon as proc-macro crates are able to export
+// anything other than proc-macros.
+#[cfg(feature = "metric-instrumentation")]
+const TOTAL_STR: &str = "total";
+#[cfg(feature = "metric-instrumentation")]
+const SUCCESS_STR: &str = "success";
+const WSV_STRING: &str = "WorldStateView";
```

Comment:
wouldn't you rather name it `iroha_core::wsv::WorldStateView`? This should be more robust

Reply:
Good example. That precise string would break in `iroha_core`, because the fully qualified path would be `crate::wsv::WorldStateView`, and it would also break if we moved `WSV` to `iroha_data_model`.